src/vs/workbench/contrib/notebook/browser/diff/diffComponents.ts

```diff
@@ -1598,7 +1600,26 @@ export class ModifiedElement extends AbstractElementRenderer {
 			modified: modifiedTextModel
 		});
 
-		this._editor!.restoreViewState(this.cell.getSourceEditorViewState() as editorCommon.IDiffEditorViewState);
+		const handleViewStateChange = () => {
+			this._editorViewStateChanged = true;
+		};
+
+		const handleScrollChange = (e: editorCommon.IScrollEvent) => {
+			if (e.scrollTopChanged || e.scrollLeftChanged) {
+				this._editorViewStateChanged = true;
```

Comment:
Should this variable be set to false again later, or does the whole object get disposed relatively quickly?

Reply:
It gets disposed when it's moved out of view. It's view instead of view model.